src/client/texturesource.cpp

```diff
@@ -1848,58 +1856,105 @@ bool TextureSource::generateImagePart(std::string_view part_of_name,
 
 #undef CHECK_DIM
 
-/*
-	Calculate the color of a single pixel drawn on top of another pixel.
 
-	This is a little more complicated than just video::SColor::getInterpolated
-	because getInterpolated does not handle alpha correctly.  For example, a
-	pixel with alpha=64 drawn atop a pixel with alpha=128 should yield a
-	pixel with alpha=160, while getInterpolated would yield alpha=96.
-*/
-static inline video::SColor blitPixel(const video::SColor src_c, const video::SColor dst_c, u32 ratio)
+namespace {
+
+/// Draw a source color on top of a destination color
```

Comment:
Does not specify how alpha is handled. The previous description was more helpful in this case.

Reply:
I have added back the previous description.

Furthermore, I've noticed a rounding inaccuracy in the alpha valua calculation: If the bottom pixel has alpha 254, the result should be fully opaque iff the top pixel has an alpha value of at least 128. Adding 127 before the division solves the inaccuracy I think:
```
lua -e 'function newa(src_a, dst_a) new_a = (src_a * 255 + (255 - src_a) * dst_a + 127) / 255  print(("src_a: %d, dst_a: %d, mixed: %f, new alpha: %d"):format(src_a, dst_a, new_a, math.floor(new_a))) end  newa(64, 128)  newa(128, 254)  newa(127, 254)'
src_a: 64, dst_a: 128, mixed: 160.372549, new alpha: 160
src_a: 128, dst_a: 254, mixed: 255.000000, new alpha: 255
src_a: 127, dst_a: 254, mixed: 254.996078, new alpha: 254
```